### PR TITLE
Add ESLint

### PR DESCRIPTION
**Pull request by dav-is**
Added Eslint. Rules still need some discussion

https://github.com/flarum/core/issues/879

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,24 +1,32 @@
...
+var eslint = require('gulp-eslint');
 var concat = require('gulp-concat');
 var babel = require('gulp-babel');
 var cached = require('gulp-cached');
 var remember = require('gulp-remember');
 var order = require('gulp-order');
 var streamqueue = require('streamqueue');
 var file = require('gulp-file');
+var gulpIf = require('gulp-if');
+var gulpTap = require("gulp-tap");
```
Comment (franzliedke) [inline index.js]: Please use single quotes here for consistency.

File: index.js [hunk 2]
```diff
@@ -1,24 +1,32 @@
...
 var babelCore = require('babel-core');
 
 function handleError(e) {
   console.log(e.toString());
   this.emit('end');
 }
+function isFixed (file) {
```
Comment (franzliedke) [inline index.js]: Please remove the space before the opening parenthesis.

**Review by franzliedke — CHANGES_REQUESTED** [review]
Please fix these minor formatting issues and then squash all your commits together.

File: package.json [hunk 3]
```diff
@@ -25,12 +25,53 @@
...
+          "/^initProps$/",
+          "static-methods",
+          "/^content$/",
+          "/^view$/",
+          "/^config$/",
+          "everything-else",
+          "/^onsubmit$/",
+          "/^onerror$/"
+        ]
+      }]
```
Comment (franzliedke) [inline package.json:70]: Can you explain what all of these do? Especially the `order` array isn't particularly self-explanatory.

**Comment by franzliedke** [comment]
@tobscure You'll have to take care of this one.

File: package.json (hunk 3 above)
Comment (dav-is) [reply package.json:70]: The order is the order methods must come in classes. I looked for the most common pattern for methods and came up with this order. As for the rest, the documentation explains it better than I could.

**Comment by dav-is** [comment]
One of the rules that I wasn't sure to turn on or off was defining methods as statics if they don't use `this`.

**Comment by franzliedke** [comment]
This package is now obsolete, in favor of flarum-webpack-config, right? Let's archive this then... @tobscure

**Pull request closed without merging**